topics/command-tips.md

```diff
@@ -0,0 +1,88 @@
+# Command tips
+
+An @array-reply of strings, command tips are meant to provide additional information about commands for clients and proxies.
+This infromation should help clients and proxies to determine how to handle this command when Redis is in clustering mode.
+Unlike the command flags (the third element in `COMMAND`'s reply), which are internal in nature (Redis itself uses them internally), command tips are just exposed via `COMMAND` and Redis does not use them.
+
+The tips are arbirary strings, but in order to create some sort of concensus, here is a list of suggested tips:
+
+## Boolean tips
+
+Tips that either exist or not, no value is associated with them.
+
+- **blocking**
+- **random-output**
+- **random-order-output**
+
+### blocking
+
+If present, this command has the potential to block (but not always, see `block_keyword` below)
+
+### random-output
+
+The output is not deterministic, that is, the same command with the same arguments, with the same key-space, may have different results. (e.g. `RANDOMKEY`, `SPOP`)
+
+### random-order-output
+
+The output is deterministic, but comes in random order (e.g. `HGETALL`, `SMEMBERS`)
+
+## Key-value tips
+
+Tips that are associated with a value.
+
+- **block_keyword** 
+- **request_policy**
+- **reply_policy**
+
+### block_keyword
+
+Identify commands which the potential to block, only is a specific arguement was given: For example, `XREAD` and `XREADGROUP` have the `blocking` tip, but they will only block if `BLOCK` was given. We will add the `block_keyword:<word>` tip, and the proxy should search `<word>` search in `argv` to determine if the command may indeed block.
+This could result in false-positive if that keyword is a keyname or some other form of free-text, but this tip best-effort: there's little harm in mistaking a non-blocking command with a blocking one (but mistaking in the other direction is unacceptable).
+
+### request_policy
```

Comment:
Nothing here says `cluster` and I think it should, to set the right context.
And another nit: we always use the term "command", not "request".

Reply:
"request" is used as a mirror for "reply", I'm open to suggestions... "command" instead of "request", but what will replace "reply"?

Reply:
(regardless, i will mention the word cluster)

Reply:
i don't think the term "command" is good.
"command" is what's being executed by redis.
i think "request" is the intent of the client app, which may need to be propagated to multiple shards and aggregate their response.
bottom line, i'm ok with the current term. i'm not strictly against "command" but i must say i don't like it.

Reply:
I like request more than command, but also don't feel that strongly.

Reply:
will will change reply -> response, keep request